core/services/pipeline/orm.go

```diff
@@ -151,23 +153,47 @@ func (o *orm) HealthReport() map[string]error {
 	return map[string]error{o.Name(): o.Healthy()}
 }
 
-func (o *orm) CreateSpec(pipeline Pipeline, maxTaskDuration models.Interval, qopts ...pg.QOpt) (id int32, err error) {
-	q := o.q.WithOpts(qopts...)
+func (o *orm) Transact(ctx context.Context, fn func(ORM) error) error {
+	//TODO wrap the function to close the ORM?
+	return sqlutil.Transact(ctx, func(tx sqlutil.DataSource) ORM {
+		return o.withDataSource(tx)
+	}, o.ds, nil, fn)
```

Comment:
This is an odd case. Typically our ORMs are just lightweight, stateless wrappers that can be created and then dropped for garbage collection, but this one implements the Service interface and launches background routines. We should probably just be calling `Close()` from here when the tx is complete.